Mage.Sets/src/mage/cards/m/MaskwoodNexus.java

```diff
@@ -81,31 +84,42 @@ public boolean apply(Game game, Ability source) {
         }
         // Creature cards you own that aren't on the battlefield
         // in graveyard
-        for (UUID cardId : controller.getGraveyard()) {
-            Card card = game.getCard(cardId);
-            if (card != null && card.isCreature(game)) {
-                game.getState().getCreateMageObjectAttribute(card, game).getSubtype().setIsAllCreatureTypes(true);
-            }
-        }
+        controller.getGraveyard().stream().map(game::getCard).filter(Objects::nonNull).filter(card -> card.isOwnedBy(controller.getId()) && card.isCreature(game))
+                .map(card -> game.getObject(card.getId())).filter(Objects::nonNull)
+                .map(CardUtil::getObjectParts).forEach(parts ->
+                        parts.stream().filter(Objects::nonNull).forEach(objectId ->{
+                            MageObject mageObject = game.getObject(objectId);
+                            game.getState().getCreateMageObjectAttribute(mageObject, game).getSubtype().setIsAllCreatureTypes(true);
+                        }));
+
         // on Hand
-        for (UUID cardId : controller.getHand()) {
-            Card card = game.getCard(cardId);
-            if (card != null && card.isCreature(game)) {
-                game.getState().getCreateMageObjectAttribute(card, game).getSubtype().setIsAllCreatureTypes(true);
-            }
-        }
+        controller.getHand().stream().map(game::getCard).filter(Objects::nonNull).filter(card -> card.isOwnedBy(controller.getId()) && card.isCreature(game))
```

Comment:
Lots of duplicated code here. I know the original had duplicated code, but with the changes it's even worse.

Please try to streamline it e.g. by adding cards from all relevant zones to one set and by factoring out the common filter.

Reply:
Done